packages/x-license/src/utils/licenseScope.ts

```diff
@@ -1,3 +1,20 @@
 export const LICENSE_SCOPES = ['pro', 'premium'] as const;
+export const PRODUCT_SCOPES = ['data-grid', 'date-pickers', 'charts', 'tree-view', null] as const;
```

Comment:
When do we need the `null` here?

Reply:
its there because `regex.match` can potentially return null. This is likely not going to happen in our case, but to make it type safe without using assertion this is one way.
Or we use `export type ProductScope = (typeof PRODUCT_SCOPES)[number] | null;` probably makes more sense.